pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -258,11 +259,11 @@ func (builder *ListenerBuilder) buildVirtualOutboundListener(
 		isTransparentProxy = proto.BoolTrue
 	}
 
-	tcpProxyFilter := newTCPProxyOutboundListenerFilter(push, node)
+	fallthroughNetworkFilters := buildFallthroughNetworkFilters(push, node)
 
 	filterChains := []*listener.FilterChain{
 		{
-			Filters: []*listener.Filter{tcpProxyFilter},
+			Filters: fallthroughNetworkFilters,
```

Comment:
in this function, after the ipTablesListener is built, add the TLS inspector to the listener. Else the forward_downstream_sni wont work. Something like this

```diff
@@ -291,9 +306,19 @@ func (builder *ListenerBuilder) buildVirtualOutboundListener(
                Address:          util.BuildAddress(actualWildcard, uint32(push.Mesh.ProxyListenPort)),
                Transparent:      isTransparentProxy,
                UseOriginalDst:   proto.BoolTrue,
+               ListenerFilters: []*listener.ListenerFilter{
+                       {Name: xdsutil.TlsInspector},
+               },
                FilterChains:     filterChains,
                TrafficDirection: core.TrafficDirection_OUTBOUND,
        }
+       ipTablesListener.ListenerFiltersTimeout = gogo.DurationToProtoDuration(push.Mesh.ProtocolDetectionTimeout)
+       if ipTablesListener.ListenerFiltersTimeout != nil {
+               ipTablesListener.ContinueOnListenerFiltersTimeout = true
+       }
```

I would probably go as far as checking if the egress proxy is set and only then add the TLS inspector